algorithmic_efficiency/data_utils.py

```diff
@@ -7,28 +9,63 @@
 from torch.utils.data import DistributedSampler
 from torch.utils.data import Sampler
 
+from algorithmic_efficiency import spec
+
 
-def shard_numpy_ds(xs):
+def shard_numpy_ds(
```

Comment:
Maybe we should rename this function to something like `shard_and_maybe_pad_ds()`? I guess the full descriptive name `convert_to_numpy_and_shard_and_maybe_pad_ds()` is a bit long haha

Reply:
Yeah I like that suggestion, I'll include this change in a follow-up PR.